cmd/client/util.go

```diff
@@ -99,6 +99,38 @@ func parseBytes(s string) (interface{}, error) {
 	return bytes, nil
 }
 
+func parseStringMap(s string) (interface{}, error) {
+	m := make(map[string]string)
+	for _, pair := range strings.Split(s, ";") {
+		colon := strings.Index(pair, ":")
+		if colon < 0 {
+			return nil, fmt.Errorf("%s is not a valid key/value pair in the form key:value", pair)
+		}
+
+		k := pair[0:colon]
+		v := pair[colon+1:]
+		m[k] = v
+	}
+	return m, nil
+}
+
+// add to dictionary
+func add2Dict(dictionary map[int32]string, s string) int32 {
```

Comment:
Given the context in which this is used, I didn't care about the redundancy in that case. But just to be anal, I added a little loop to discover and reuse existing dictionary entries.